v2/client.go

```diff
@@ -859,11 +936,12 @@ func (c *client) Subscribe(ctx context.Context, streamName string, handler Handl
 // passed through MessageOptions, if any. If a partition or Partitioner is not
 // provided, this defaults to the base partition. This partition determines the
 // underlying NATS subject that gets published to.  To publish directly to a
-// spedcific NATS subject, use the low-level PublishToSubject API.
+// specific NATS subject, use the low-level PublishToSubject API.
 //
 // If the AckPolicy is not NONE, this will synchronously block until the ack is
 // received. If the ack is not received in time, ErrAckTimeout is returned. If
-// AckPolicy is NONE, this returns nil on success.
+// AckPolicy is NONE, this returns nil on success. A FailedPrecondition status
```

Comment:
I think rather than returning a grpc error we should return a Readonly error. We could change `ErrEndOfReadonlyPartition` to `ErrReadonlyPartition` and return that here for instance.